qiskit/circuit/library/basis_change/qft.py

```diff
@@ -218,7 +218,21 @@ def inverse(self) -> 'QFT':
         Returns:
             The inverted circuit.
         """
-        inverted = super().inverse()
+
+        if self.name in ('qft', 'iqft'):
+            name = 'qft' if self._inverse else 'iqft'
+        else:
+            name = self.name + '_dg'
+
+        inverted = self.copy(name=name)
+        inverted._data = []
+
+        from qiskit.circuit.parametertable import ParameterTable
```

Comment:
Only because it's a bit of hack (qft shouldn't have to manage it's own parameter table). Looking now though, this could be avoided if we set `inverted.data = []` instead of `inverted._data = []`. I'll take care of that.

Reply:
Never mind, `inverted.data = []` fails with `AttributeError: can't set attribute`. Maybe because `BlueprintCircuit` overrides the `data` property but not also the setter?

Reply:
```python
super()._invalidate()
self._data = []
```
should work I think, the blueprint circuit resets the parameter table upon `_invalidate`. (Then also the `_invalidate` method of QFT can be removed).

Reply:
:+1: I'll open an issue

Reply:
https://github.com/Qiskit/qiskit-terra/issues/4353